### ISSUE TITLE
False positive and maybe-false negative with cascade_invocations with cascaded invocations on difference references

### ISSUE THREAD

**Issue by Cyp** [dart-lang/linter#3240]
**Describe the issue**
False positive with `cascade_invocations` on different references.

Also, a bonus maybe-false negative with uncascaded invocations on equal references, if allowed to assume that the getter's return value doesn't change.

**To Reproduce**
```dart
const a = 1;
const b = 2;
a.bitLength
  ..abs()
  ..ceil();
b.bitLength  // “Cascade consecutive method invocations on the same reference.”, false positive.
  ..abs()
  ..ceil();
b.bitLength.abs();  // No lint, maybe-false negative.
b.bitLength.ceil();  // No lint, maybe-false negative (if assuming that `abs()` didn't change the value of `b.bitLength`).
```

**Expected behavior**
No lint (although in this very specific example, a lint about `abs` and `ceil` having no side effects would be reasonable).

**Additional context**
The linter seems to think `a.something` and `b.something` are the same object, which is not in general even likely to be true.

**Comment by srawlins** [comment]
I fixed a very very related bug in https://github.com/dart-lang/linter/issues/1323 many moons ago. In that fix, I added a TODO which is basically the "false negative" half of your bug.

I just addressed the "false positive" half of this bug here: https://github.com/dart-lang/linter/pull/3713.

**Comment by karabanovbs** [comment]
any updates?

**Comment by srawlins** [comment]
I did try to fix this and found it extremely difficult. I mean to chat with @bwilkerson about it.

**Comment by srawlins** [comment]
Quick drive by: I see in the tests for `join_return_with_assignment` that they may have a mechanism for this in the impl there. The `unJoinablePropertyAccessReturn` and `joinablePropertyAccessReturn` test cases demonstrate this.
